middleware/swagger-metadata.js

```diff
@@ -225,7 +225,7 @@ var processOperationParameters = function (swaggerMetadata, pathKeys, pathMatch,
 
   debug('  Processing Parameters');
 
-  async.map(_.reduce(parameters, function (requestParsers, parameter) {
+  var parsers = _.reduce(parameters, function (requestParsers, parameter) {
```

Comment:
You cannot simply call `multiPartParser.fields([{name: paramName, maxCount: 1}])` for each file.
Lets say you have `file1` and `file2`, when the handler for `file1` is called, it will see data for a file called `file2` which will throw an "unexpected field" error.
Also, if you send `multipart/form-data`without file in it, you need to use `multiPartParser.array()` to handle it, and to do so, you need to know if your endpoint wait for file or not.
This two constraints together led me to that implementation.